spec/mongoid/reloadable_spec.rb

```diff
@@ -356,6 +356,71 @@
       end
     end
 
+    context "when embeds_many documents are cleared and reassigned" do
+
+      context "when update_embedded_after_nil feature flag is set" do
+        around do |example|
+          saved_flag = Mongoid::update_embedded_after_nil
+          Mongoid::update_embedded_after_nil = true
+          begin
+            example.run
+          ensure
+            Mongoid::update_embedded_after_nil = saved_flag
+          end
+        end
+
+        let(:contractor) do
+          Contractor.new(name: 'contractor')
+        end
+
+        let(:building) do
+          Building.create!
+        end
+
+        before do
+          building.contractors << contractor
+          building.contractors.clear
```

Comment:
This flow should have intermediate assertions, for example that after `clear` there aren't any contractors.

Reply:
done